File: civitas/src/GUI/Controlador.java
package GUI;

import civitas.*;
import static java.lang.System.exit;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Scanner;
import GUI.Respuestas;

public class Controlador {
    private CivitasJuego juego;
    private CivitasView vista;
    
    public Controlador(CivitasJuego juego, CivitasView vista) {
        this.juego = juego;
        this.vista = vista;
    }
    
    public void juega() {
        vista.setCivitasJuego(juego);
        
        while(!juego.finalDelJuego()) {
            OperacionesJuego operacion = juego.siguientePaso();
            vista.actualizarVista();
            vista.mostrarSiguienteOperacion(operacion);
            if(operacion != civitas.OperacionesJuego.PASAR_TURNO) {
                vista.mostrarEventos();
            }
            if(!juego.finalDelJuego()) {
                switch(operacion) {
                    case COMPRAR:
                        Respuestas res = vista.comprar();
                        if(res == Respuestas.SI) {
                            juego.comprar();
                        }
                        juego.siguientePasoCompletado(operacion);
                        break;
                        
                    case GESTIONAR:
                        vista.gestionar();
                        vista.getGestion();
                        vista.getPropiedad();
                        OperacionInmobiliaria oi = new OperacionInmobiliaria(GestionesInmobiliarias.values()[vista.getGestion()], vista.getPropiedad());
                        
                        switch(GestionesInmobiliarias.values()[vista.getGestion()]) {
                            case VENDER:
                                if(juego.getJugadorActual().getNombrePropiedades() != null) {
                                    juego.vender(oi.getNumPropiedad());
                                    break;
                                }
                            case HIPOTECAR:
                                if(juego.getJugadorActual().getNombrePropiedades() != null) {
                                    juego.hipotecar(oi.getNumPropiedad());
                                    break;
                                }
                            case CANCELAR_HIPOTECA:
                                if(juego.getJugadorActual().getNombrePropiedades() != null) {
                                   juego.cancelarHipoteca(oi.getNumPropiedad());
                                   break;
                                }
                            case CONSTRUIR_CASA:
                                juego.construirCasa(oi.getNumPropiedad());
                                break;
                            case CONSTRUIR_HOTEL:
                                juego.construirHotel(oi.getNumPropiedad());
                                break;
                            default:
                                juego.siguientePasoCompletado(operacion);
                               break;
                        }
                        break;
                    case SALIR_CARCEL:
                        SalidasCarcel salida = vista.salirCarcel();
                        switch(salida) {
                            case PAGANDO:
                                juego.salirCarcelPagando();
                                break;
                            case TIRANDO:
                                juego.salirCarcelTirando();
                                break;
                        }
                }
            }else{
                ArrayList<Jugador> ranking = juego.ranking();
                for(int i = 0; i < ranking.size(); i++) {
                    System.out.println(ranking.get(i).getNombre() + "\n");
                }
                vista.actualizarVista();
            }
        }
    }
    
}

File: civitasRuby/lib/controlador.rb

module Civitas
  
  require_relative 'civitas_juego'
  require_relative 'operacion_inmobiliaria'
  require_relative 'operaciones_juego'
  require_relative 'respuestas'
  require_relative 'gestiones_inmobiliarias'
  require_relative 'salidas_carcel'

  
  class Controlador
    
    def initialize(juego, vista)
      @juego = juego
      @vista = vista
    end
    
    def juega
      @vista.set_civitas_juego(@juego)
      while !@juego.final_del_juego
        @vista.actualizar_vista
        @vista.pausa
        operacion = @juego.siguiente_paso
        @vista.mostrar_siguiente_operacion(operacion)
        if operacion != Civitas::OperacionesJuego::PASAR_TURNO
          @vista.mostrar_eventos
        end
        if !@juego.final_del_juego
          case operacion
          when Civitas::OperacionesJuego::COMPRAR
            res = @vista.comprar
            if res == Civitas::Respuestas::SI
              @juego.comprar
            end
            @juego.siguiente_paso_completado(operacion)
          when Civitas::OperacionesJuego::GESTIONAR
            @vista.gestionar
            @vista.iGestion
            @vista.iPropiedad
            lista = [Civitas::GestionesInmobiliarias::VENDER, 
              Civitas::GestionesInmobiliarias::HIPOTECAR, 
              Civitas::GestionesInmobiliarias::CANCELAR_HIPOTECA, 
              Civitas::GestionesInmobiliarias::CONSTRUIR_CASA, 
              Civitas::GestionesInmobiliarias::CONSTRUIR_HOTEL, 
              Civitas::GestionesInmobiliarias::TERMINAR]
            oi = OperacionInmobiliaria.new(lista[@vista.iGestion], 
              @vista.iPropiedad)
            case lista[@vista.iGestion]
            when Civitas::GestionesInmobiliarias::VENDER
              if !@juego.get_jugador_actual.get_nombre_propiedades != nil
                @juego.vender(oi.num_propiedad)
              end
            when Civitas::GestionesInmobiliarias::HIPOTECAR
              if !@juego.get_jugador_actual.get_nombre_propiedades != nil
                @juego.hipotecar(oi.num_propiedad)
              end
            when Civitas::GestionesInmobiliarias::CANCELAR_HIPOTECA
              if !@juego.get_jugador_actual.get_nombre_propiedades != nil
                @juego.cancelar_hipoteca(oi.num_propiedad)
              end
            when Civitas::GestionesInmobiliarias::CONSTRUIR_CASA
                @juego.construir_casa(oi.num_propiedad)
            when Civitas::GestionesInmobiliarias::CONSTRUIR_HOTEL
                @juego.construir_hotel(oi.num_propiedad)
            else
              @juego.siguiente_paso_completado(operacion)
            end
            
          when Civitas::OperacionesJuego::SALIR_CARCEL
            salida = @vista.salir_carcel
            case salida
            when Civitas::SalidasCarcel::PAGANDO
              @juego.salir_carcel_pagando
            when Civitas::SalidasCarcel::TIRANDO
              @juego.salir_carcel_tirando
            end
            @juego.siguiente_paso_completado(operacion)
          end
        else
          lista = Array.new
          lista = @juego.ranking
          lista.reverse
          for i in 0...lista.size
            puts lista.at(i).nombre + "\n"
          end
        end
      end
    end
    
  end
end
